### PR TITLE
docs: new quickstart guide

### LINKED ISSUE #440: devenv.sh script is missing
**Is this a request for help?**: No

---

**TYPE: ISSUE**
---

According to the documentation, there should be a `scripts/devenv.sh` script for building aks-engine binaries. It definitely was in acs-engine before
https://github.com/Azure/aks-engine/blob/master/docs/tutorials/quickstart.md

**What version of aks-engine?**:
---
0.29.1 / master

Either documentation should be updated (like we just support PS, install ps on linux and go away), or missing script should be added and maintained.

### PR DESCRIPTION
**Pull request by bacongobbler**
This PR streamlines the quickstart guide to

1. Install the Azure CLI
1. Install AKS Engine
1. Create a resource group and a service principal
1. `aks-engine deploy`

The streamlining contains all the steps required to create a cluster using AKS Engine in one document.

Fixes #440 

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

### CODE REVIEW

File: docs/howto/troubleshooting.md [hunk 1]
```diff
@@ -155,3 +155,33 @@ read and **write** permissions to the target Subscription.
...
+
+```sh
+az vm show -g <resource group of cluster> -n <name of Master or agent VM> --query tags
+```
+
+    Sample JSON out of this command is shown below. This command can also be used to check the aks-engine version which was used to create the cluster
+
+```json
+{
+  "acsengineVersion": "v0.15.0",
```
Comment (CecileRobertMichon) [inline docs/howto/troubleshooting.md]: this is aksEngineVersion now

File: docs/howto/troubleshooting.md [hunk 2]
```diff
@@ -155,3 +155,33 @@ read and **write** permissions to the target Subscription.
...
+  "creationSource": "acsengine-k8s-master-22116803-0",
```
Comment (CecileRobertMichon) [inline docs/howto/troubleshooting.md]: `aksengine` now

File: docs/tutorials/quickstart.md [hunk 3]
```diff
@@ -1,190 +1,240 @@
...
+
+```console
+$ aks-engine deploy --subscription-id 51ac25de-afdg-9201-d923-8d8e8e8e8e8e \
+    --auth-method cli \
+    --dns-prefix contoso-apple \
+    --resource-group contoso-apple \
+    --location westus2 \
+    --api-model examples/kubernetes.json \
+    --set servicePrincipalProfile.clientId="47a62f0b-917c-4def-aa85-9b010455e591" \
+    --set servicePrincipalProfile.secret="26054d2b-799b-448e-962a-783d0d6f976b"
```
Comment (CecileRobertMichon) [inline docs/tutorials/quickstart.md]: these might get detected as exposed secrets by audits, let's redact them (even though they are fake)

File: docs/community/developer-guide.md [hunk 4]
```diff
@@ -23,6 +23,107 @@ Make sure you have read and understood the main CONTRIBUTING guide:
...
 https://github.com/Azure/aks-engine/blob/master/CONTRIBUTING.md
 
+## Build AKS Engine from Source
+
+### Docker Development Environment
+
+The easiest way to start hacking on AKS Engine is to use a Docker-based environment. If you already have Docker installed then you can get started with a few commands.
+
+* Windows (PowerShell): `.\scripts\devenv.ps1`
+* Linux/OSX (bash): `./scripts/devenv.sh`
```
Comment (CecileRobertMichon) [inline docs/community/developer-guide.md]: I wonder if we should mention `make dev` here? Personally it's what I use and it's pretty convenient

File: docs/tutorials/quickstart.md [hunk 5]
```diff
@@ -1,190 +1,240 @@
...
-Use "aks-engine [command] --help" for more information about a command.
+Once that's done, we need to create a [service principal][sp] for the Kubernetes cluster so it can talk to the any resources that are a part of the same resource group.
+
+```console
+$ az ad sp create-for-rbac --role="Contributor" --scopes="/subscriptions/51ac25de-afdg-9201-d923-8d8e8e8e8e8e/resourceGroups/contoso-apple"
+{
+  "appId": "47a62f0b-917c-4def-aa85-9b010455e591",
+  "displayName": "azure-cli-2019-01-11-22-22-06",
+  "name": "http://azure-cli-2019-01-11-22-22-06",
+  "password": "26054d2b-799b-448e-962a-783d0d6f976b",
```
Comment (CecileRobertMichon) [inline docs/tutorials/quickstart.md]: same here for password

**Comment by bacongobbler** [comment]
all comments addressed. Once e2e passes I'll merge

**Comment by CecileRobertMichon** [comment]
@bacongobbler you have a conflict but otherwise I think this is ready to go. It won't run E2E because it's a docs-only change.

File: docs/tutorials/quickstart.md [hunk 6]
```diff
@@ -1,186 +1,240 @@
...
-3. **azuredeploy.parameters.json**: the parameters file holds a series of custom variables which are used in various locations throughout `azuredeploy.json`.
-4. **certificate and access config files**: orchestrators like Kubernetes require certificates and additional configuration files (e.g. Kubernetes apiserver certificates and kubeconfig).
+The `aks-engine deploy` command automates creation of a Service Principal, Resource Group and SSH key for your cluster. If operators need more control or are interested in the individual steps see the ["Long Way" section below](#aks-engine-the-long-way).
 
-### Generate Templates
+**NOTE:** AKS Engine creates a _cluster_; it _doesn't_ create an Azure Kubernetes Service resource. Clusters that you create using the `aks-engine` command (or ARM templates generated by the `aks-engine` command) won't show up as AKS resources, for example when you run `az aks list`.  Think of `aks-engine` as the, er, engine which AKS uses to create clusters: you can use the same engine yourself, but AKS won't know about the results.
```
Comment (tariq1890) [inline docs/tutorials/quickstart.md]:
```suggestion
**NOTE:** AKS Engine creates a _cluster_; it _doesn't_ create an Azure Kubernetes Service(AKS) resource. Clusters that you create using the `aks-engine` command (or ARM templates generated by the `aks-engine` command) won't show up as AKS resources, for example when you run `az aks list`.  Think of `aks-engine` as the, er, engine which AKS uses to create clusters: you can use the same engine yourself, but AKS won't know about the results.
```

Comment (tariq1890) [reply docs/tutorials/quickstart.md]: This is a great note. Thanks for adding this.

File: docs/tutorials/quickstart.md [hunk 7]
```diff
@@ -1,186 +1,240 @@
...
-  4. Type `make build` to build the project
-  5. Type `./bin/aks-engine` to see the command line parameters
+* To enable the optional network policy enforcement using calico, you have to set the parameter during this step according to this [guide](../topics/features.md#optional-enable-network-policy-enforcement-using-calico)
+* To enable the optional network policy enforcement using cilium, you have to set the parameter during this step according to this [guide](../topics/features.md#optional-enable-network-policy-enforcement-using-cilium)
+
+Now we can deploy the files `azuredeploy.json` and `azuredeploy.parameters.json` using either the Azure CLI or PowerShell.
+
+Using the CLI:
+
+```text
```
Comment (tariq1890) [inline docs/tutorials/quickstart.md]:
```suggestion
```sh
```

File: docs/tutorials/quickstart.md [hunk 8]
```diff
@@ -1,186 +1,240 @@
...
 
-New-AzureRmResourceGroup `
-    -Name <RESOURCE_GROUP_NAME> `
-    -Location <LOCATION>
+We will also need to generate an SSH key. When creating VMs, you will need an SSH RSA key for SSH access. Use the following articles to create your SSH RSA Key:
+
+1. Windows - https://www.digitalocean.com/community/tutorials/how-to-create-ssh-keys-with-putty-to-connect-to-a-vps
+1. Mac and Linux - https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/
+
+Next, we'll create a resource group. A resource group is a container that holds related resources for an Azure solution. In Azure, you logically group related resources such as storage accounts, virtual networks, and virtual machines (VMs) to deploy, manage, and maintain them as a single entity. In this case, we want to deploy, manage and maintain the whole Kubernetes cluster as a single entity.
```
Comment (tariq1890) [inline docs/tutorials/quickstart.md]:
```suggestion
Next, we'll create a resource group. A resource group is a container (not to be confused with Docker containers.) that holds related resources for an Azure solution. In Azure, you logically group related resources such as storage accounts, virtual networks, and virtual machines (VMs) to deploy, manage, and maintain them as a single entity. In this case, we want to deploy, manage and maintain the whole Kubernetes cluster as a single entity.
```

Comment (tariq1890) [reply docs/tutorials/quickstart.md]: Super nit: Although I am not comfy with this definition, I understand that this is Microsoft's definition of a resource group and I feel this is disambiguation would be useful for a newbie.

File: docs/tutorials/quickstart.md (hunk 6 above)
Comment (CecileRobertMichon) [reply docs/tutorials/quickstart.md]: I'd add a space before the `(AKS)`

Comment (tariq1890) [reply docs/tutorials/quickstart.md]: Thanks!

Comment (CecileRobertMichon) [reply docs/tutorials/quickstart.md]: nice use of the suggested change feature :)

File: docs/tutorials/quickstart.md (hunk 8 above)
Comment (bacongobbler) [reply docs/tutorials/quickstart.md]: changed terminology to "logical grouping unit".

Comment (tariq1890) [reply docs/tutorials/quickstart.md]: Perfect!

File: docs/community/developer-guide.md [hunk 9]
```diff
@@ -23,6 +23,107 @@ Make sure you have read and understood the main CONTRIBUTING guide:
...
+
+This script mounts the `aks-engine` source directory as a volume into the Docker container, which means you can edit your source code in your favorite editor on your machine, while still being able to compile and test inside of the Docker container. This environment mirrors the environment used in the AKS Engine continuous integration (CI) system.
+
+When the script `devenv.ps1` or `devenv.sh` completes, you will be left at a command prompt.
```
[the remark this answers is not in the document]
Comment (bacongobbler) [reply docs/community/developer-guide.md]: done!

File: docs/tutorials/quickstart.md (hunk 6 above)
Comment (bacongobbler) [reply docs/tutorials/quickstart.md]: done!

File: docs/tutorials/quickstart.md (hunk 7 above)
Comment (bacongobbler) [reply docs/tutorials/quickstart.md]: switched to `console` to match the rest of the docs. Thanks!

**Pull request merged**